build_tools/hack/env_check/device.py

```diff
@@ -0,0 +1,631 @@
+from __future__ import annotations
+from .find_tools import FindPython, FindMSVC
+from .utils import *
+import platform, re, sys, os
+
+
+class Device:
```

Comment:
"Device" is a pretty generic class and file name. In this domain (ML libraries, firmware, etc.) it is also a very overloaded term.

How about "SystemInformation"?

```suggestion
class SystemInformation:
``